packages/client-nuxt/src/index.ts

```diff
@@ -0,0 +1,244 @@
+import type { NuxtApp } from 'nuxt/app';
+import {
+  useAsyncData,
+  useFetch,
+  useLazyAsyncData,
+  useLazyFetch,
+} from 'nuxt/app';
+
+import type { Client, Config, RequestOptions } from './types';
+import {
+  buildUrl,
+  createConfig,
+  createInterceptors,
+  getParseAs,
+  mergeConfigs,
+  mergeHeaders,
+  setAuthParams,
+} from './utils';
+
+type ReqInit = Omit<RequestInit, 'body' | 'headers'> & {
+  body?: any;
+  headers: ReturnType<typeof mergeHeaders>;
+};
+
+export const createClient = (config: Config = {}): Client => {
+  let _config = mergeConfigs(createConfig(), config);
+
+  const getConfig = (): Config => ({ ..._config });
+
+  const setConfig = (config: Config): Config => {
+    _config = mergeConfigs(_config, config);
+    return getConfig();
+  };
+
+  const interceptors = createInterceptors<
+    Request,
+    Response,
+    unknown,
+    RequestOptions
+  >();
+
+  const clientRequest: Client['clientRequest'] = ({
+    asyncDataOptions,
+    composable,
+    fetchOptions,
+    key,
+    requestFetch,
+    url,
+    ...opts
+  }) => {
+    const fetchFn = requestFetch ?? $fetch;
+    const baseUrl = 'https://petstore3.swagger.io/api/v3';
```

Comment:
was looking to test out the PR, however, looks like this is hardcoded so it's not possible to test out alternative URLs. is this expected, or?

Reply:
It's a work in progress, tons of stuff left to do. I'll ping you when it's more ready

Reply:
Hi @lrstanley, want to give it a try? Would love your thoughts

```sh
npm i https://pkg.pr.new/hey-api/openapi-ts/@hey-api/client-nuxt@1519
```